### PR TITLE
Add support for x86 architecture

### PR DESCRIPTION
**Pull request by Jasdriel**
I add the json files and also add the options for choose the 32 bit for each compiler (msvc, clang, gcc), If you could guide more if I am missing something or if some logic also needs to be changed

### CODE REVIEW

File: Extension/bin/msvc.32.darwing.msvc.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (bobbrow) [inline Extension/bin/msvc.32.darwing.msvc.json]: This file name has a spelling error in it `darwing` -> `darwin`

File: Extension/c_cpp_properties.schema.json [hunk 2]
```diff
@@ -78,6 +78,9 @@
                             "msvc-x64",
                             "gcc-x64",
                             "clang-x64",
+                            "msvc-x86",
```
Comment (bobbrow) [inline Extension/c_cpp_properties.schema.json:88]: We're going to need to update the cpptools API too.

File: Extension/ui/settings.html [hunk 3]
```diff
@@ -510,6 +512,9 @@
             <option value="msvc-x64">msvc-x64</option>
             <option value="gcc-x64">gcc-x64</option>
             <option value="clang-x64">clang-x64</option>
+            <option value="msvc-x86">msvc-x86</option>
```
Comment (bobbrow) [inline Extension/ui/settings.html]: Minor issue.  Let's sort these:

clang-x86
clang-x64
gcc-x86
gcc-x64
msvc-x86
msvc-x64

**Review by bobbrow — CHANGES_REQUESTED** [review]

File: Extension/c_cpp_properties.schema.json (hunk 2 above)
Comment (Jasdriel) [reply Extension/c_cpp_properties.schema.json:88]: ok, also sort the elements like in the settings.html?

File: Extension/ui/settings.html [hunk 4]
```diff
@@ -499,7 +499,9 @@
         <div class="section-title">IntelliSense mode</div>
         <div class="section-text">
             The IntelliSense mode used by the IntelliSense engine. <code>msvc-x64</code> maps to Visual Studio mode with 64-bit pointer sizes. 
-            <code>clang-x64</code> maps to CLang mode with 64-bit pointer sizes. <code>gcc-x64</code> maps to GCC mode with 64-bit pointer sizes. 
+            <code>clang-x64</code> maps to CLang mode with 64-bit pointer sizes. <code>gcc-x64</code> maps to GCC mode with 64-bit pointer sizes.
```
Comment (michelleangela) [inline Extension/ui/settings.html]: Let's simplify description of `IntelliSense` to:

> The IntelliSense mode to use that maps to an architecture-specific variant of MSVC, gcc, or Clang. If not set or if set to `${default}`, the extension will choose the default for that platform. Windows defaults to `msvc-x64`, Linux defaults to `gcc-x64`, and macOS defaults to `clang-x64`. 

Need to also update https://code.visualstudio.com/docs/cpp/c-cpp-properties-schema-reference

Comment (Jasdriel) [reply Extension/ui/settings.html]: Got it

File: Extension/c_cpp_properties.schema.json (hunk 2 above)
Comment (bobbrow) [reply Extension/c_cpp_properties.schema.json:88]: it isn't necessary for the schema.

File: Extension/ui/settings.html [hunk 5]
```diff
@@ -498,18 +498,20 @@
     <div class="section">
         <div class="section-title">IntelliSense mode</div>
         <div class="section-text">
-            The IntelliSense mode used by the IntelliSense engine. <code>msvc-x64</code> maps to Visual Studio mode with 64-bit pointer sizes. 
-            <code>clang-x64</code> maps to CLang mode with 64-bit pointer sizes. <code>gcc-x64</code> maps to GCC mode with 64-bit pointer sizes. 
-            The <code>${default}</code> mode will choose the default for that platform. 
+            The IntelliSense mode to use that maps to an architecture-specific variant of MSVC, gcc, or Clang. 
+            If not set or if set to ${default}, the extension will choose the default for that platform.
```
Comment (michelleangela) [inline Extension/ui/settings.html]: Add `<code>` style around `${default}` like below:

`<code>${default}</code>`

Comment (Jasdriel) [reply Extension/ui/settings.html]: Thank you

File: Extension/c_cpp_properties.schema.json (hunk 2 above)
Comment (bobbrow) [inline Extension/c_cpp_properties.schema.json:88]: Can we update the description here to match the settings.html?

**Review by bobbrow — CHANGES_REQUESTED** [review]

File: Extension/c_cpp_properties.schema.json [hunk 6]
```diff
@@ -72,12 +72,15 @@
                         }
                     },
                     "intelliSenseMode": {
-                        "description": "If set, it overrides the default mode used by the IntelliSense engine. Windows defaults to msvc-x64, Linux defaults to gcc-x64, and Mac defaults to clang-x64.",
+                        "description": "The IntelliSense mode to use that maps to an architecture-specific variant of MSVC, gcc, or Clang. If not set or if set to ${default}, the extension will choose the default for that platform. Windows defaults to msvc-x64, Linux defaults to gcc-x64, and macOS defaults to clang-x64. Select a specific IntelliSense mode to override the ${default} mode.",
```
Comment (bobbrow) [inline Extension/c_cpp_properties.schema.json]: I would remove the last line. `Select a specific IntelliSense mode to override the ${default} mode.`

**Review by bobbrow — APPROVED** [review]

**Review by michelleangela — APPROVED** [review]

**Pull request merged**